controllers/cluster_controller.go

```diff
@@ -373,6 +373,10 @@ func (r *ClusterReconciler) preClusterProviderReconcile(ctx context.Context, log
 		if err := validations.ValidateManagementEksaVersion(mgmt, cluster); err != nil {
 			return controller.Result{}, err
 		}
+
+		if err := validateEksaRelease(ctx, r.client, cluster); err != nil {
```

Comment:
why is this only validated for workload clusters? shouldn't this validation be for all clusters?

Reply:
Good catch. I forgot users can upgrade management clusters through FLC now. I'll change it so we validate for all clusters.